### DISCUSSION TITLE
Can role support authority?

### DISCUSSION THREAD

**Discussion by meidea** [zitadel/zitadel#4478] [Ideas]
User with `ROLE_ADMIN` role have the authorities to `READ`,`DELETE`,`WRITE`,`UPDATE`.

User with `ROLE_MANAGER` role have the authorities to `READ`, `WRITE` and `UPDATE`.

User with role `ROLE_USER` has authority to `READ` only.

**Comment by mffap** [comment]
hi @meidea I don't understand your question, could you elaborate? Are you referring to roles in projects (which you can choose freely) or ZITADEL's Manager roles (can be configured in self-hosting atm.)?

**Reply by meidea** [reply]
@mffap roles in projects；

eg:

The authority is written in the code：
```java
@PostMapping("/user/list")
@PreAuthorize({"USER_READ"})
public List<User> list(@RequestBody @Valid Query query) {
    return null;
}

```

Create roles based on organizational structure;

**Reply by mffap** [reply]
At the moment the roles are non-hierarchical. Although we see some demand for introducing a grouping / structure for roles, see the discussion here: https://github.com/zitadel/zitadel/discussions/2134

**Comment by meidea** [comment]
eg:

![create_role](https://user-images.githubusercontent.com/19621448/193713978-b54cfcb9-69f3-4502-a262-547e5b13c00a.jpeg)
